In [4]:
#Importation de la bibliothèque PANDAS
import pandas as pd

In [5]:
#Créons un dictionnaire de données 
pays= {
    
    "Pays" : ["Burkina", "Cote d'ivoir", "Mali", "Niger"],
    "Capitale": ["Ouagadougou", "Yamoussokro","Bamako", "Niamey"],
    "Superficie": [274200, 322462, 1240190, 1186408],
    "PIB": [15, 52, 19, 9]
}

In [7]:
#Transformons notre dictionnaire en DataFrame

pays=pd.DataFrame(pays)

In [8]:
pays

Pays     Capitale  Superficie  PIB
0       Burkina  Ouagadougou      274200   15
1  Cote d'ivoir  Yamoussokro      322462   52
2          Mali       Bamako     1240190   19
3         Niger       Niamey     1186408    9

In [9]:
print(pays.columns)

Index(['Pays', 'Capitale', 'Superficie', 'PIB'], dtype='object')


# Manipulation de DataFrame

# IMPORTATION DE DONNÉES

In [5]:
pays = pd.read_excel('pays_afrique_complet.xlsx')
pays.head(10)
#pays.tail()

pays drapeau     capitale  Population  Superficie      pib  \
0  Afrique du Sud      🇿🇦     Pretoria    61525521     1219912  419.946   
1         Algérie      🇩🇿        Alger    44487616     2381741  193.601   
2          Angola      🇦🇴       Luanda    35981281     1246700  175.540   
3           Bénin      🇧🇯   Porto-Novo    13301694      114763   17.140   
4        Botswana      🇧🇼     Gaborone     2317233      581726   15.780   
5    Burkina Faso      🇧🇫  Ouagadougou    20835401      274400   16.220   
6         Burundi      🇧🇮       Gitega    11865821       27834    2.780   
7        Cameroun      🇨🇲      Yaoundé    27744989      475442   40.130   
8         Comores      🇰🇲       Moroni      936833        2236    1.340   
9           Congo      🇨🇬  Brazzaville     5400000      342000   14.490   

      langue monnaie  
0    Anglais     ZAR  
1      Arabe     DZD  
2  Portugais     AOA  
3   Français     XOF  
4    Anglais     BWP  
5   Français     XOF  
6   Français     BIF  
7   Français     XAF  
8   Français     KMF  
9   Français     XAF

In [6]:
pays.shape

(54, 8)

In [7]:
pays.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54 entries, 0 to 53
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   pays        54 non-null     object 
 1   drapeau     54 non-null     object 
 2   capitale    54 non-null     object 
 3   Population  54 non-null     int64  
 4   Superficie  54 non-null     int64  
 5   pib         54 non-null     float64
 6   langue      54 non-null     object 
 7   monnaie     54 non-null     object 
dtypes: float64(1), int64(2), object(5)
memory usage: 3.5+ KB


In [8]:
pays.dtypes

pays           object
drapeau        object
capitale       object
Population      int64
Superficie      int64
pib           float64
langue         object
monnaie        object
dtype: object

In [10]:
#Récupere un enregistrement

pays.iloc[5]['capitale'] # affiche la 6e ligne

'Ouagadougou'

In [12]:
pays.iloc[-2]['Population']

19610769

In [13]:
pays_peupler= pays[pays['Population']>= 100000000]
pays_peupler

pays drapeau     capitale  Population  Superficie     pib  \
10  Congo (Rép. dém.)      🇨🇩     Kinshasa   111859928     2345860   55.35   
13             Égypte      🇪🇬     Le Caire   102334000     1001450  404.10   
15           Éthiopie      🇪🇹  Addis-Abeba   110871031     1127127   80.87   
36            Nigéria      🇳🇬        Abuja   219463862      923768  440.80   

       langue monnaie  
10   Français     CDF  
13      Arabe     EGP  
15  Amharique     ETB  
36    Anglais     NGN

In [14]:
#Affichons les pays luxophone (Portugais)
pays_portugais=pays[pays['langue']=='Portugais']
pays_portugais 

pays drapeau  capitale  Population  Superficie      pib  \
2                 Angola      🇦🇴    Luanda    35981281     1246700  175.540   
21         Guinée-Bissau      🇬🇼    Bissau     2061000       36125    1.630   
23              Cap-Vert      🇨🇻     Praia      583255        4033    1.704   
33            Mozambique      🇲🇿    Maputo    27233789      801590   15.780   
40  São Tomé et Príncipe      🇸🇹  São Tomé      211122        1001    0.400   

       langue monnaie  
2   Portugais     AOA  
21  Portugais     XOF  
23  Portugais     CVE  
33  Portugais     MZN  
40  Portugais     STD

In [15]:
pays_cfa_fr= pays[(pays['monnaie']=='XOF') & (pays['langue']=='Français')]
pays_cfa_fr

pays drapeau      capitale  Population  Superficie     pib  \
3           Bénin      🇧🇯    Porto-Novo    13301694      114763  17.140   
5    Burkina Faso      🇧🇫   Ouagadougou    20835401      274400  16.220   
11  Côte d'Ivoire      🇨🇮  Yamoussoukro    29389150      322462  61.502   
30           Mali      🇲🇱        Bamako    21726000     1241238  17.163   
35          Niger      🇳🇪        Niamey    22772361     1267000  14.920   
41        Sénégal      🇸🇳         Dakar    16209125      196722  27.630   
50           Togo      🇹🇬          Lomé     8608444       56785   8.413   

      langue monnaie  
3   Français     XOF  
5   Français     XOF  
11  Français     XOF  
30  Français     XOF  
35  Français     XOF  
41  Français     XOF  
50  Français     XOF

In [16]:
#Ordonnée les pays par superficie

paysorted=pays.sort_values(by='Superficie', ascending=False)
paysorted.head()

pays drapeau   capitale  Population  Superficie      pib  \
1             Algérie      🇩🇿      Alger    44487616     2381741  193.601   
10  Congo (Rép. dém.)      🇨🇩   Kinshasa   111859928     2345860   55.350   
45             Soudan      🇸🇩   Khartoum    43120843     1886068   34.330   
27              Libye      🇱🇾    Tripoli     6890535     1759540   42.820   
49              Tchad      🇹🇩  N'Djamena    15946882     1284000   11.780   

      langue monnaie  
1      Arabe     DZD  
10  Français     CDF  
45     Arabe     SDG  
27     Arabe     LYD  
49  Français     XAF

# Statistique descriptive avec Pandas

In [17]:
#Stats



print("PIB moyen: ", pays['pib'].mean())
print("La médiane du PIB  : ", pays['pib'].median())
print("PIB max : ", pays['pib'].max())
print("PIB min : ", pays['pib'].min())

plage_pib=pays['pib'].max()- pays['pib'].min()

print("Plage du PIB :", plage_pib)
print("Variance du PIB  : ", pays['pib'].var())
print("Ecart-type du PIB :", pays['pib'].std())

PIB moyen:  50.88592592592593
La médiane du PIB  :  15.35
PIB max :  440.8
PIB min :  0.4
Plage du PIB : 440.40000000000003
Variance du PIB  :  9875.81782674913
Ecart-type du PIB : 99.37714941951761


In [18]:
pays.describe()

Population    Superficie         pib
count  5.400000e+01  5.400000e+01   54.000000
mean   2.561317e+07  5.569692e+05   50.885926
std    3.782588e+07  6.040768e+05   99.377149
min    9.846200e+04  4.550000e+02    0.400000
25%    2.859055e+06  6.052375e+04    5.464250
50%    1.359371e+07  3.322310e+05   15.350000
75%    2.897811e+07  8.991805e+05   40.087500
max    2.194639e+08  2.381741e+06  440.800000

In [19]:
pays.describe(include=object)

pays drapeau  capitale   langue monnaie
count               54      54        54       54      54
unique              54      54        54        7      42
top     Afrique du Sud      🇿🇦  Pretoria  Anglais     XOF
freq                 1       1         1       20       8

In [36]:
#Afficher le nombre de pays

pays.count()

pays          54
drapeau       54
capitale      54
Population    54
Superficie    54
pib           54
langue        54
monnaie       54
dtype: int64

In [44]:
pays['pays'].count()

54

In [54]:
#Fonction Idxmax renvoie les indice des lignes maximale de chaque colone et on recupere avec ILOC 

m=pays['Population'].idxmax()
pays.iloc[m]

pays.iloc[m]['pays']

'Nigéria'

In [41]:
#pays le plus peuplé 

pays[pays['Population']== pays['Population'].max()]['pays']

36    Nigéria
Name: pays, dtype: object

Nigéria est le pays le plus peuplé de l'Afrique avec une population de 219463862 habitants !

In [38]:
#pays le plus peuplé 

pays[pays['Population']== pays['Population'].min()]['pays']

42    Seychelles
Name: pays, dtype: object

Par contre Seychelles est le moins peuplé avec une population de 98462	 habitants

In [23]:

print("Population moyenne: ", pays['Population'].mean())
print("La médiane de la Population  : ", pays['Population'].median())

Population moyenne:  25613167.444444444
La médiane de la Population  :  13593709.0


In [60]:
#langue la plus parlé

pays['langue'].value_counts()                                 #nous renvois chaque langue avec le nombre d'occurence 
pays['langue'].value_counts().idxmax()             #récupére là  le nom de la langue


'Anglais'

In [58]:
#Population moyenne des pays anglophones

pays_anglais=pays[pays['langue']=='Anglais']

print("La moyenne de la population anglaise  : ", pays_anglais ['Population'].mean())

La moyenne de la population anglaise  :  28906502.2


In [63]:
#PIB  le plus et le moins élvevé des pays de XOF


pays_xof=pays[pays['monnaie']=='XOF']
print ( pays_xof['pib'].max())
print( pays_xof['pib'].min())

61.502
1.63


In [64]:
#PIB  le plus élvevé des pays de XOF
pays_xof[pays_xof['pib']== pays_xof['pib'].max()]

pays drapeau      capitale  Population  Superficie     pib  \
11  Côte d'Ivoire      🇨🇮  Yamoussoukro    29389150      322462  61.502   

      langue monnaie  
11  Français     XOF

In [66]:
#Comparaison de PIB moyen entre les pays anglophone et francophone
pays_franco=pays[pays['langue']=='Français']


if  pays_franco['pib'].mean() < pays_anglais['pib'].mean() :
    print("Les pays anglophones ont le pib moyen le plus élévé ")
else :
    print("Les pays francophones ont le pib moyen le plus élévé ")

Les pays anglophones ont le pib moyen le plus élévé 


In [29]:
#Comparaison de PIB  total entre la zone XOF(Afrique de l'ouest) et XAF (Afrique central)
pays_xaf=pays[pays['monnaie']=='XAF']

pib_total_xof= sum(pays_xof['pib'])
pib_total_xaf= sum(pays_xaf['pib'])


print ("Le PIB total pour la zone XOF : ", pib_total_xof)
print ("Le PIB total pour la zone XAF : ", pib_total_xaf)

Le PIB total pour la zone XOF :  164.618
Le PIB total pour la zone XAF :  98.03900000000002


In [30]:
#Comparaison de Population  entre la zone XOF(Afrique de l'ouest) et XAF (Afrique central)

pop_total_xof= sum(pays_xof['Population'])
pop_total_xaf= sum(pays_xaf['Population'])

print ("La Population total pour la zone XOF : ", pop_total_xof)
print ("Le Population total pour la zone XAF : ", pop_total_xaf)


La Population total pour la zone XOF :  134903175
Le Population total pour la zone XAF :  58332011


L'Afrique de l'ouest est plus peuplé de l'Afrique centrale

In [31]:
#  Comparaison de PIB /Habitant  entre la zone XOF(Afrique de l'ouest) et XAF (Afrique central)

pib_h_xof= pib_total_xof/pop_total_xof
pib_h_xaf= pib_total_xaf/pop_total_xaf

print("Le PIB par habitant pour la zone XOF : ", pib_h_xof)
print("Le PIB par habitant pour la zone XAF : ", pib_h_xaf)

Le PIB par habitant pour la zone XOF :  1.2202677957727828e-06
Le PIB par habitant pour la zone XAF :  1.6807066706477857e-06


Le PIB/Habitant est légèrement élevé en centrale qu'en Afrique de l'Ouest.